qcfractal/interface/client.py

```diff
@@ -241,15 +245,37 @@ def add_options(self, opt_list, full_return=False):
 
     ### Collections section
 
-    def get_collections(self, collection_list, full_return=False):
+    def list_collections(self, collection_type=None):
+
+        query = {}
+        if collection_type is not None:
+            query = {"collection": collection_type.lower()}
+
+        payload = {"meta": {"projection": {"name": True, "collection":True}}, "data": query}
+        r = self._request("get", "collection", payload)
+
+        if collection_type is None:
+            ret = defaultdict(list)
+            for entry in r.json()["data"]:
+                ret[entry["collection"]].append(entry["name"])
+            return dict(ret)
+        else:
+            return [x["name"] for x in r.json()["data"]]
+
 
-        payload = {"meta": {}, "data": collection_list}
+    def get_collection(self, collection_type, collection_name, full_return=False):
+
+        payload = {"meta": {}, "data": [(collection_type.lower(), collection_name)]}
         r = self._request("get", "collection", payload)
 
         if full_return:
             return r.json()
         else:
-            return r.json()["data"]
+            # If nothing found
+            if len(r.json()["data"]):
+                return collection_factory(r.json()["data"][0], client=self)
```

Comment:
Should this thrown an error if there are more than 1 collection? Otherwise there will be ambiguous returns which may or may not be deterministic.

Reply:
It isn't currently possible to get more than one return here at the moment due to the unique requirements of the index.

Reply:
This is fine then!